src/libraries/System.Private.Xml/src/System/Xml/Core/XmlWriterAsync.cs

```diff
@@ -599,5 +599,21 @@ private async Task WriteLocalNamespacesAsync(XPathNavigator nsNav)
                 await WriteAttributeStringAsync("xmlns", prefix, XmlReservedNs.NsXmlNs, ns).ConfigureAwait(false);
             }
         }
+
+        public async ValueTask DisposeAsync()
+        {
+            await DisposeAsyncCore().ConfigureAwait(false);
+            Dispose(false);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual ValueTask DisposeAsyncCore()
+        {
+            if (WriteState != WriteState.Closed)
+            {
+                Dispose(true);
+            }
+            return default;
```

Comment:
can this be more explicit? Does default mean SuccessfullyCompleted/Error or something else?

Reply:
This definitely not error, but it could mean SuccessfullyCompleted or it was `already Disposed/Closed` state, so i think its better to keep default